doc/design/draft/subscription.md

```diff
@@ -0,0 +1,180 @@
+# Subscription
+
+## Summary
+
+In this document we analyse how many and what kind of chunks will be provided
+to a subscriber on subscription. The answer depends on the number of publishers,
+how many samples were send in the past and how large the requested _HistorySize_
+is.
+
+The goal is to craft and implement a design which is unable to deadlock,
+massively slow down (caused by deadlock discovery) or crash RouDi.
+
+## Terminology
+
+- **ChunkHistory** the last N samples which the publishers of a service have
+   published
+- **HistorySize** the number of samples the subscriber will receive at most on
+   subscription
+- **DeliveryTime** the point in time where the publisher writes the samples in
```

Comment:
It is important to note that time points are only relevant if they are monotonic and, depending on clock synchronization, may be meaningless across process boundaries.

Reply:
@MatthiasKillat here I had a simple `uint64_t` number in mind which will increase with every delivery and is stored in the shared memory. 

This would be a simplistic and efficient way to sort the samples after there delivery time - we do not require a real time format for that, this would fit all the requirements.
But in the end it feels like a sequence number - maybe we just define the sequence number that way and say that it corresponds with the delivery time.